pkgs/by-name/au/autoenv/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "autoenv";
+  version = "0-unstable-2024-10-16";
+
+  src = fetchFromGitHub {
+    owner = "hyperupcall";
+    repo = "autoenv";
+    rev = "90241f182d6a7c96e9de8a25c1eccaf2a2d1b43a";
+    hash = "sha256-vZrsMPhuu+xPVAww04nKyoOl7k0upvpIaxeMrCikDio=";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share
+    cp -R $src $out/share/autoenv
+    chmod +x $out/share/autoenv
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Per-directory shell environments sourced from .env file";
+    homepage = "https://github.com/hyperupcall/autoenv";
+    license = lib.licenses.mit;
+    mainProgram = "autoenv";
```

Comment:
There is no binary at $out/bin/autoenv. You can create a symlink to $out/share/activate.sh in installPhase.

Reply:
That is what the module activation already does.

Reply:
If you set `mainProgram = x`, there must be a file at `$out/bin/x`. I would suggest link $out/bin/autoenv to $out/share/activate.sh.

Reply:
I think `mainProgram` doesn't make sense for this package. Unlike e.g: `direnv`, there is no executable for autoenv, is there?

Reply:
Oh right. Doesn't really make sense to execute the activation script like that. In that case you can just omit mainProgram from meta.

Reply:
I understand, will just remove that line then!

Reply:
Applied all suggestions and squashed the commits.